app/code/Magento/Catalog/Block/Adminhtml/Category/Tab/Product.php

```diff
@@ -110,37 +117,41 @@ protected function _addColumnFilterToCollection($column)
     }
 
     /**
-     * @return Grid
+     * @inheritdoc
      */
     protected function _prepareCollection()
     {
         if ($this->getCategory()->getId()) {
             $this->setDefaultFilter(['in_category' => 1]);
         }
-        $collection = $this->_productFactory->create()->getCollection()->addAttributeToSelect(
-            'name'
-        )->addAttributeToSelect(
-            'sku'
-        )->addAttributeToSelect(
-            'visibility'
-        )->addAttributeToSelect(
-            'status'
-        )->addAttributeToSelect(
-            'price'
-        )->joinField(
-            'position',
-            'catalog_category_product',
-            'position',
-            'product_id=entity_id',
-            'category_id=' . (int)$this->getRequest()->getParam('id', 0),
-            'left'
-        );
-        $storeId = (int)$this->getRequest()->getParam('store', 0);
-        if ($storeId > 0) {
-            $collection->addStoreFilter($storeId);
-        }
-        $this->setCollection($collection);
 
+        if (!$this->getCollection()) {
+            $collection = $this->_productFactory->create()->getCollection()->addAttributeToSelect(
+                'name'
+            )->addAttributeToSelect(
+                'sku'
+            )->addAttributeToSelect(
+                'visibility'
+            )->addAttributeToSelect(
+                'status'
+            )->addAttributeToSelect(
+                'price'
+            )->joinField(
+                'position',
+                'catalog_category_product',
+                'position',
+                'product_id=entity_id',
+                'category_id=' . (int)$this->getRequest()->getParam('id', 0),
+                'left'
+            );
```

Comment:
I know that's about the preferences, but I personally hate these line-breaks.
```suggestion
            $collection = $this->_productFactory->create()->getCollection()
                ->addAttributeToSelect('name')
                ->addAttributeToSelect('sku')
                ->addAttributeToSelect('visibility')
                ->addAttributeToSelect('status')
                ->addAttributeToSelect('price')
                ->joinField(
                    'position',
                    'catalog_category_product',
                    'position',
                    'product_id=entity_id',
                    ['category_id' => (int)$this->getRequest()->getParam('id')],
                    'left'
                );
```